applications/body_pose_estimation/metadata.json

```
@@ -34,14 +34,50 @@
 			"hardware": [
 				{
 					"name": "camera",
-					"description": "This application requires a Video4Linux (V4L2) compatible device as input.",
-					"required": true
+					"description": "This application requires a Video4Linux (V4L2) compatible device as input."
 				}
 			]
 		},
-		"run": {
-			"command": "python3 <holohub_app_source>/body_pose_estimation.py",
-			"workdir": "holohub_bin"
+		"modes": {
+			"default": {
+				"description": "Run with camera input (default)",
+				"hardware": ["camera"],
+				"run": {
+					"command": "python3 <holohub_app_source>/body_pose_estimation.py",
+					"workdir": "holohub_bin"
+				}
+			},
+			"replayer": {
+				"description": "Run with pre-recorded video (no camera required)",
+				"run": {
+					"command": "python3 <holohub_app_source>/body_pose_estimation.py",
+					"workdir": "holohub_bin",
+					"args": "--source replayer"
+				}
+			},
+			"dds": {
+				"description": "Run with DDS video stream input",
+				"build": {
+					"op_depends": [
```

Updated internal doc to suggest `target_depends`

```suggestion
					"target_depends": [
```